gen/src/main.rs

```diff
@@ -231,21 +240,51 @@ fn main() {
     eprintln!("All bindings generated!");
 }
 
-fn git_checkout(rev: &str) {
-    // Delete any generated files from previous versions.
+fn git_init() {
+    // Clone the linux kernel source repo if necessary. Ignore exit code as it will be non-zero in
+    // case it was already cloned.
+    // Use a treeless partial clone to save disk space and clone time.
+    // See https://github.blog/2020-12-21-get-up-to-speed-with-partial-clone-and-shallow-clone/ for
+    // more info on partial clones.
+    // Note: this is not using the official repo
+    // git://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git but the github fork as the
+    // server of the official repo doesn't recognize filtering.
+    Command::new("git")
+        .arg("clone")
+        .arg("https://github.com/torvalds/linux.git")
+        .arg("--filter=tree:0")
+        .arg("--no-checkout")
+        .status()
+        .unwrap();
```

Comment:
What happens if a user does a build in a tree that already has gen/linux cloned in it?

Reply:
This command will have a non-zero exit code. The `.unwrap()` only checks if the command succeeded to fork/exec. The exit code is ignored. Other commands have an assert on `.success()`.

Reply:
Ah, I missed that. I'm uncomfortable ignoring errors in general, because `git clone` could fail for other reasons. Could this instead do a check, such as checking that gen/linux/.git exists, and skip the `git clone` in that case?